pkg/volume/emptydir/empty_dir.go

```diff
@@ -212,7 +212,12 @@ func (ed *emptyDir) SetUpAt(dir string, mounterArgs volume.MounterArgs) error {
 		if ed.medium == v1.StorageMediumMemory && !notMnt {
 			return nil
 		} else if ed.medium == v1.StorageMediumDefault {
-			return nil
+			// Further check dir exists
+			if _, err := os.Stat(dir); err == nil {
+				return nil
+			}
+			// This situation should not happen. In this case, print a warning for it.
```

Comment:
Should we remove readydir too?

Reply:
I think we can. Since the current order of operations in setup() and tearDown() guarantee that readyfile shouldn't exist if vol-dir does not exist. I think this situation happens only if someone deletes vol-dir out of band. In that case, kubelet restart is needed to recover, irrespective of we delete the readydir here or not. Isnt it?

Reply:
y, no matter readyfile exist or not, we want to continue to recover. I think if in the case we allow concurrent volume setup, delete ready file is a must to avoid another process see ready exist and path exist and then consider setup is done while the first one is still in the middle of setting up permission and ownership. Although in operation executor, we don't allow the same volume same pod to start concurrent volume setup calls, this should not be an issue. But I agree delete ready file might be safer in case we have bugs on allowing concurrent operations.